src/kudu/consensus/consensus_meta.cc

```diff
@@ -84,6 +85,15 @@ void ConsensusMetadata::set_voted_for(const string& uuid) {
   DFAKE_SCOPED_RECURSIVE_LOCK(fake_lock_);
   DCHECK(!uuid.empty());
   pb_.set_voted_for(uuid);
+
+  // Populate previous vote information.
+  DCHECK(pb_.has_current_term());
+  PreviousVotePB prev_vote;
+  prev_vote.set_candidate_uuid(uuid);
+  prev_vote.set_election_term(pb_.current_term());
+
+  // TODO(ritwikyadav) : Put an upper bound on the memory utilized.
```

Comment:
this is a reasonably big gap. We can't let the previous vote history grow for ever.

Reply:
Yes. I think I have it figured out on how to delete it. Wanted to finalize this part first. Will add more commits to the PR soon.